appsettings/src/tree/AppSettingsTreeItem.ts

```diff
@@ -74,6 +74,19 @@ export class AppSettingsTreeItem extends AzExtParentTreeItem {
         }
     }
 
+    static async createAppSettingsTreeItem(context: IActionContext, parent: AzExtParentTreeItem, clientProvider: AppSettingsClientProvider, extensionPrefix: string, options?: AppSettingsTreeItemOptions): Promise<AppSettingsTreeItem> {
```

Comment:
I know I already looked at this, but I just thought of this--

You should probably make the constructor private so that people don't accidentally create the app settings with a raw new constructor since if they did, it wouldn't have the expected behavior.